src/common/components/DataTable/DataTable.tsx

```diff
@@ -123,6 +135,19 @@ export const DataTable = <D extends Record<string, unknown>>({
     [pageIndex, pageSize, paginationEnabled, pagination?.onPageChange, pagination?.onPageSizeChange],
   );
 
+  // When switching between pages via the Paginator, the table's pageIndex will always be 1 less than pagination.page.
+  // However, if the user uses search or a filter, then pagination.page will be less than or equal to the pageIndex.
+  // In this case, we need to reset the table's pageIndex to 0. This will cause the pagination numbering and next/previous
+  // buttons to be updated and the result will be consistent with the current data set.
+  useEffect(() => {
+    if (pagination) {
+      if (pagination.page <= pageIndex) {
+        dispatch({ type: 'filter or search was used' });
```

Comment:
Not really worth using a dispatch here. We can use `gotoPage` instead from `useTable`. Other than that, nice work!

![image](https://user-images.githubusercontent.com/2287977/188525053-4998e7e5-a2e5-4575-963d-bcdb9a5e450c.png)